tests/self-tests.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+if [[ $EUID -ne 0 ]]; then
+    >&2 echo "Must be run as root"
+    exit 1
+fi
+
+set -e;
+
+TESTS_DIR="$(dirname "${BASH_SOURCE[0]}")";
+DIR="$( cd $TESTS_DIR >/dev/null && pwd )"
+BPFTRACE_RUNTIME_TEST_EXECUTABLE=${BPFTRACE_EXECUTABLE:-$DIR/../src/bpftrace};
+export BPFTRACE_RUNTIME_TEST_EXECUTABLE;
+
+echo "===================="
+echo "bpftrace --info:"
+echo "===================="
+"${BPFTRACE_RUNTIME_TEST_EXECUTABLE}" --info;
+
+find $TESTS_DIR/self -type f -a -name \*.bt -print0 | xargs -0L1 "${BPFTRACE_RUNTIME_TEST_EXECUTABLE}" --test;
```

Comment:
Stupid question, if one test fails will it stop the whole suite? If not, does a subsequent passing test change the exit status?

Reply:
xargs will continue running all commands, but will return a non-zero exit code if any of the commands returns a non-zero exit code.

```
       xargs exits with the following status:

              0      if it succeeds

              123    if any invocation of the command exited with status
                     1–125

              124    if the command exited with status 255

              125    if the command is killed by a signal

              126    if the command cannot be run

              127    if the command is not found

              1      if some other error occurred.

       Exit codes greater than 128 are used by the shell to indicate that
       a program died due to a fatal signal.
```